extensions/Corbnorb/noise.js

```diff
@@ -0,0 +1,3459 @@
+            j1 -= yNSign * this._PrimeY;
+          } else {
+            z1 += zNSign;
+            b = b + az0 + az0;
+            k1 -= zNSign * this._PrimeZ;
+          }
+
+          if (b > 1) {
+            b -= 1;
+            let bbbb = b * b * (b * b);
+            let xo, yo, zo;
+            if (outGradOnly) {
+              let hash = this._HashR3(seed, i1, j1, k1) & (255 << 2);
+              xo = this._RandVecs3D[hash];
+              yo = this._RandVecs3D[hash | 1];
+              zo = this._RandVecs3D[hash | 2];
+            } else {
+              let hash = this._HashR3(seed, i1, j1, k1);
+              let index1 = hash & (63 << 2);
+              let index2 = (hash >> 6) & (255 << 2);
+              let xg = this._Gradients3D[index1];
+              let yg = this._Gradients3D[index1 | 1];
+              let zg = this._Gradients3D[index1 | 2];
+              let value = x1 * xg + y1 * yg + z1 * zg;
+              let xgo = this._RandVecs3D[index2];
+              let ygo = this._RandVecs3D[index2 | 1];
+              let zgo = this._RandVecs3D[index2 | 2];
+              xo = value * xgo;
+              yo = value * ygo;
+              zo = value * zgo;
+            }
+            vx += bbbb * xo;
+            vy += bbbb * yo;
+            vz += bbbb * zo;
+          }
+
+          if (l === 1) break;
+
+          ax0 = 0.5 - ax0;
+          ay0 = 0.5 - ay0;
+          az0 = 0.5 - az0;
+
+          x0 = xNSign * ax0;
+          y0 = yNSign * ay0;
+          z0 = zNSign * az0;
+
+          a += 0.75 - ax0 - (ay0 + az0);
+
+          i += (xNSign >> 1) & this._PrimeX;
+          j += (yNSign >> 1) & this._PrimeY;
+          k += (zNSign >> 1) & this._PrimeZ;
+
+          xNSign = -xNSign;
+          yNSign = -yNSign;
+          zNSign = -zNSign;
+
+          seed += 1293373;
+        }
+
+        coord.x += vx * warpAmp;
+        coord.y += vy * warpAmp;
+        coord.z += vz * warpAmp;
+      };
+
+      if (arguments.length === 7) {
+        R2(
+          arguments[0],
+          arguments[1],
+          arguments[2],
+          arguments[3],
+          arguments[4],
+          arguments[5],
+          arguments[6]
+        );
+      }
+
+      if (arguments.length === 8) {
+        R3(
+          arguments[0],
+          arguments[1],
+          arguments[2],
+          arguments[3],
+          arguments[4],
+          arguments[5],
+          arguments[6],
+          arguments[7]
+        );
+      }
+    }
+  }
+
+  class Vector2 {
+    /**
+     * 2d Vector
+     * @param {number} x
+     * @param {number} y
+     */
+    constructor(x, y) {
+      this.x = x;
+      this.y = y;
+    }
+  }
+
+  class Vector3 {
+    /**
+     * 3d Vector
+     * @param {number} x
+     * @param {number} y
+     * @param {number} z
+     */
+    constructor(x, y, z) {
+      this.x = x;
+      this.y = y;
+      this.z = z;
+    }
+  }
+
+  Scratch.extensions.register(new PerlinNoise());
+})((window.Scratch = window.Scratch || {}));
```

Comment:
Scratch is always in globalThis / globally accessible

Reply:
not sure what youre asking me to change

Reply:
You don't need to do the whole window.Scratch = window.Scratch || {}